opsdroid/connector/slack/connector.py

```diff
@@ -0,0 +1,376 @@
+"""A connector for Slack."""
+import json
+import logging
+import os
+import re
+import ssl
+
+import aiohttp
+import certifi
+import opsdroid.events
+from emoji import demojize
+from opsdroid.connector import Connector, register_event
+from opsdroid.connector.slack.create_events import SlackEventCreator
+from opsdroid.connector.slack.events import Blocks, EditedBlocks
+from slack_sdk.errors import SlackApiError
+from slack_sdk.socket_mode.aiohttp import SocketModeClient
+from slack_sdk.socket_mode.request import SocketModeRequest
+from slack_sdk.socket_mode.response import SocketModeResponse
+from slack_sdk.web.async_client import AsyncWebClient
+from voluptuous import Inclusive, Required
+
+_LOGGER = logging.getLogger(__name__)
+CONFIG_SCHEMA = {
+    Required("token"): str,
+    Inclusive("socket-mode", "backend"): bool,
+    Inclusive("app-token", "backend"): str,
```

Comment:
Maybe something like this:
```python
_use_app_token_msg=(
    "Please upgrade your config with a new Slack 'app-token' instead of the classic 'token'. "
    "To access the new Slack events APIs, OpsDroid needs new-style Slack app tokens. "
    "See the docs for instructions on creating the new Slack app and configuring the 'app-token'."
    "<url here>"
)
CONFIG_SCHEMA = {
    Exclusive("token", "token", msg=_use_app_token_msg): str,
    Exclusive(Required("app-token"), "token", msg=_use_app_token_msg): str,
    "socket-mode": bool,
```

I'm not sure if Exclusive and Required can be used together like that.

Reply:
```python
_use_bot_token_msg=(
    "Please upgrade your config with a new Slack 'bot-token' instead of the classic 'token'. "
    "To access the new Slack events APIs, OpsDroid needs new-style Slack app tokens. "
    "See the docs for instructions on creating the new Slack app and configuring the 'app-token'."
    "<url here>"
)
CONFIG_SCHEMA = {
    Exclusive("token", "token", msg=_use_bot_token_msg): str,
    Exclusive(Required("bot-token"), "token", msg=_use_bot_token_msg): str,
    Inclusive("socket-mode", "backend"): bool,
    Inclusive("app-token", "backend"): str,
```

Reply:
It's not possible to use `Exclusive`  that way. I have removed the `token` as required and replaced it with `bot-token` this way if `token` exists in the config it will throw an error, asking to update config